austation/code/game/gamemodes/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,304 @@
+
+
+// Gamemode is all done being set up. We have all our Vamps. We now pick objectives and let them know what's happening.
+/datum/game_mode/bloodsucker/post_setup()
+
+	// Sunlight (Creating Bloodsuckers manually will check to create this, too)
+	check_start_sunlight()
+
+	// Vamps
+	for(var/datum/mind/bloodsucker in bloodsuckers)
+		// spawn() --> Run block of code but game continues on past it.
+		// sleep() --> Run block of code and freeze code there (including whoever called us) until it's resolved.
+
+		//Clean Bloodsucker Species (racist?)
+		//clean_invalid_species(bloodsucker)
+		// 			TO-DO !!!
+
+		// Add Bloodsucker Antag Datum (or remove from list on Fail)
+		if (!make_bloodsucker(bloodsucker))
+			bloodsuckers -= bloodsucker
+			bloodsucker.special_role = null
+
+	// NOTE: Hunters are done in ..() parent proc
+
+	return ..()
+
+// Checking for ACTUALLY Dead Vamps
+/datum/game_mode/bloodsucker/are_special_antags_dead()
+	// Bloodsucker not Final Dead
+	for(var/datum/mind/bloodsucker in bloodsuckers)
+		if(!bloodsucker.AmFinalDeath())
+			return FALSE
+	return TRUE
+
+
+// Init Sunlight (called from datum_bloodsucker.on_gain(), in case game mode isn't even Bloodsucker
+/datum/game_mode/proc/check_start_sunlight()
+	// Already Sunlight (and not about to cancel)
+	if (istype(bloodsucker_sunlight) && !bloodsucker_sunlight.cancel_me)
+		return
+	bloodsucker_sunlight = new ()
+
+// End Sun (last bloodsucker removed)
+/datum/game_mode/proc/check_cancel_sunlight()
+	// No Sunlight
+	if (!istype(bloodsucker_sunlight))
+		return
+	if (bloodsuckers.len <= 0)
+		bloodsucker_sunlight.cancel_me = TRUE
+		qdel(bloodsucker_sunlight)
+		bloodsucker_sunlight = null
+
+/datum/game_mode/proc/is_daylight()
+	return istype(bloodsucker_sunlight) && bloodsucker_sunlight.amDay
+
+//////////////////////////////////////////////////////////////////////////////
+
+
+/datum/game_mode/proc/can_make_bloodsucker(datum/mind/bloodsucker, datum/mind/creator, display_warning=TRUE) // Creator is just here so we can display fail messages to whoever is turning us.
+	// No Mind
+	if(!bloodsucker || !bloodsucker.key) // KEY is client login?
+		//if(creator) // REMOVED. You wouldn't see their name if there is no mind, so why say anything?
+		//	to_chat(creator, "<span class='danger'>[bloodsucker] isn't self-aware enough to be raised as a Bloodsucker!</span>")
+		return FALSE
+	// Current body is invalid
+	if(!iscarbon(bloodsucker.current) || isIPC(bloodsucker.current) || isplasmaman(bloodsucker.current))// && !ismonkey(bloodsucker.current))
+		if(display_warning && creator)
+			to_chat(creator, "<span class='danger'>[bloodsucker] isn't evolved enough or the right race to be raised as a Bloodsucker!</span>")
+		return FALSE
+	// Species Must have a HEART (Sorry Plasmabois)
+
+	//if(NOBLOOD in H.dna.species.species_traits)
+	//	if(display_warning && creator)
+	//		to_chat(creator, "<span class='danger'>[bloodsucker]'s DNA isn't compatible!</span>")
+	//	return FALSE
+
```

Comment:
```suggestion
	// Check Preferences
	if(isnewplayer(bloodsucker.current))
		if (bloodsucker.current.client)
			var/datum/preferences/P = bloodsucker.current.client.prefs
			if (NOBLOOD in P.pref_species.species_traits)
				return FALSE // sorry plasma and ipc mains
		else
			return FALSE // no client (shouldn't happen lmao)
	// Excludes plsamaman and ipcs
	else if(ishuman(bloodsucker.current))
		var/mob/living/carbon/human/H = bloodsucker.current
		if(isIPC(H) || isplasmaman(H))// && !ismonkey(bloodsucker.current))
			if(display_warning && creator)
				to_chat(creator, "<span class='danger'>[bloodsucker] isn't the right race to be raised as a Bloodsucker!</span>")
			return FALSE
	// Sieves other NOBLOOD species if they ever happen
		if(NOBLOOD in H.dna.species.species_traits)
			if(display_warning && creator)
				to_chat(creator, "<span class='danger'>[bloodsucker]'s DNA isn't compatible!</span>")
			return FALSE
	else
		if(display_warning && creator)
			to_chat(creator, "<span class='danger'>[bloodsucker] isn't evolved enough to be raised as a Bloodsucker!</span>")
		return FALSE
```

This fix is about _really_ excluding plasmeme and ipcs. On pre_setup, minds, aka "bloodsucker" in code, doesn't really have a real body like mob/living/carbon/human; it instead has (or, technically speaking, the mind's current is) mob/dead/new_player until actual setup is done; so on pre_setup, minds are attached to mob/dead/new_player, so things like isIPC() and H.dna.species doesn't work. So I decided to read clients preferences instead, thus, this kinda excessively long code is born.